diff --git a/index.ts b/index.ts
index aa13928..cb0c402 100644
--- a/index.ts
+++ b/index.ts
@@ -7,3 +7,12 @@ function makeClient(endpoint: string): PolkadotClient {
   const client = createClient(provider);
   return client;
 }
+
+// TODO: Create the main() function here
+// async fucntion main()
+
+// TODO: Then get the client by calling the makeClient function from above
+// giving as endpoint the "wss://rpc.polkadot.io"
+// TODO: Try to log the `polkadotClient` for this step in order to see the methods mentioned in the README
+
+// TODO: Call the main function
